libs/parallelism/algorithms/include/hpx/parallel/algorithms/partition.hpp

```diff
@@ -159,14 +609,15 @@ namespace hpx { namespace parallel { inline namespace v1 {
                     typename std::iterator_traits<RandIter>::difference_type;
 
                 future<RandIter> result;
+                auto last_iter = detail::advance_to_sentinel(first, last);
 
                 try
                 {
-                    difference_type size = std::distance(first, last);
+                    difference_type size = detail::distance(first, last);
```

Comment:
We talked about this for other algorithms. We should probably create a facility that does both in one loop: advancing the iterator _and_ returning the distance.

Reply:
Is this okay? https://github.com/STEllAR-GROUP/hpx/blob/b9a3820071ecf47286c58661a5086772933a5214/libs/parallelism/algorithms/include/hpx/parallel/algorithms/detail/advance_and_get_distance.hpp